source/common/access_log/access_log_manager_impl.cc

```diff
@@ -159,10 +159,12 @@ void AccessLogFileImpl::flushThreadFunc() {
         const Api::IoCallBoolResult open_result = open();
         if (!open_result.return_value_) {
           stats_.reopen_failed_.inc();
-          return;
```

Comment:
Removing this return is still not going to hit the else block below that you added, right? So why change this part? Are there any implications of calling doWrite() even if the reopen fails? It seems like we should still return here.

Reply:
The purpose of removing this return is to make flush thread not exit , so the buffer can be moved from flush_buffer_ to about_to_write_buffer_, and about_to_write_buffer_ can be drained next flush loop(with file_->isOpen() == false).I don't know why "if we failed to open file before, then simply ignore" rather than try open it every time. So I just drain buffer if reopen fail. If we have failed to open , I think there is no need to call doWrite(),just drain the buffer is fine.

Reply:
Yup OK this makes sense. Can you just add more comments here then? Thank you.

Reply:
Do you mean add some code comment?

Reply:
Yes add more code comments, though now I'm looking at this more and don't you need to set reopen_file_ back to true or not set it to false unless the reopen succeeds? I think this code probably needs a bit more refactoring and some further tests.

Reply:
If we decide to ignore open fail, reopen_file_ set to false is more appropriate.

Reply:
Wouldn't you want to try to open again on the next flush? I'm not sure this change as is makes sense.

Reply:
Maybe try to open every flush if open failed before?But I am not sure open fail too many times affect performance or not? I will do some test after my vacation.

Reply:
The flush thread only runs either on a timer or after a certain amount of data. I think it's fine to try to reopen each time it runs if the file is not yet open. This will be simpler to understand and would be a nice enhancement. I think this behavior will be confusing.

So changes that are needed:
1) That above logic changes.
2) This definitely needs a release note.
3) We might consider a runtime guard for this behavior but not sure if that is necessary or not. @alyssawilk thoughts?
4) Please add better tests for this new logic.

Thank you!

/wait

Reply:
Ok,I agress!Have done some changes.I am not sure if there any case we open file failed ,but not due to reopen ?